Server/src/lobby.erl

```diff
@@ -0,0 +1,115 @@
+	  %Confirmation that the diagram coordinator received the message
+	  receive
+	  	ok -> ok
+	  end,
+	  %Receives the result from the diagram coordinator and sends it to the client
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when a previous message has been readded to the "queue" or if there were to previous messages to step back to.
+	{previous_confirmation, Did, Message} -> 
+	  %Turns the message we want to send into binary.
+	  Format_result = io_lib:format("~p", [{Did, previous_confirmation, Message}]),
+	  %Sends it to the client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	
+	%This case happens when there is no more messages in the diagram and the user tries to simulate the next message
+	{simulation_done, Did, Message_number} -> 
+	  %Turns the result into binary
+	  Format_result = io_lib:format("~p", [{Did, simulation_finished, Message_number}]),
+      %Sends it to the client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when a message has been sent in a diagram and the info is supposed to be sent to the client
+	{message_sent, Did, From, To, Message, Message_number} -> 
+	  %Turns the result into binary
+	  %The character ~ is used as the stop character for when the client should stop reading from the tcp connection
+	  Format_result = io_lib:format("~p", [{Did, From, To, Message, Message_number}]) ++ "~",
+      %Sends it to the client
+	  send_messages(Members, [Format_result]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID);
+	  
+	%This case happens when there are messages to print to the execution log
+	{Did, print_information, Msg}->
+	  %Turns the message into binary
+	  Format_result = io_lib:format("~p", [{Did, print_information, Msg}]),
+      %Sends the result to client
+	  send_messages(Members, [Format_result ++ "~"]),
+	  loop(Creator_Socket, Password, Members, Diagrams, Lobby_ID)
+  end.
+  
+%Sends the classes of all given diagrams to the given user
+send_classes(_, []) -> done;
+send_classes(Socket, [{Diagram_ID, Class_names, _}|Diagrams]) -> 
+  Format_result = io_lib:format("~p", [{Diagram_ID, Class_names}]) ++ "~",
+  gen_tcp:send(Socket, Format_result),
+  send_classes(Socket, Diagrams).
+  
+%Finds the correct diagram from the given list
+find_diagram(_, []) -> not_created;
+find_diagram(Diagram_id,[{Diagram_id, _ ,Pid} | _]) -> Pid;
+find_diagram(Diagram_id,[_| Diagrams])  -> find_diagram(Diagram_id, Diagrams).  
+  
+%Finds the given member in the given list of members
+find_member([], _) -> not_found;
+find_member([Member|_], Member) -> found_member;
+find_member([_|Members], Member) -> find_member(Members, Member).
+
+%Sends a message to all the users in the given list
+send_messages(Members, Message) -> [gen_tcp:send(Socket, Message) || Socket <- Members].
```

Comment:
**Since 'all' (or most) calls to this method look like this:**

```
Format_result = io_lib:format("~p", [{Did, simulation_finished, Message_number}]),
send_messages(Members, [Format_result ++ "~"]),
```

**I suggest just revamping the function to include this functionality.**

```
send_messages(Members, [{Did, atom, Number}]) -> 
	send_messages(Members, [io_lib:format("~p", [{Did, atom, Number}] ++ "~"]);
send_messages(Members, Message) -> 
	[gen_tcp:send(Socket, Message) || Socket <- Members]. 
```

*** This has not been tested, and is just a suggestion to simplify many of the cases in the class.**

Reply:
This will be a bit too time consuming to change for now, we can change it if we ever get the time